### ISSUE TITLE
Standard test-and-release workflow should include master AND main

### ISSUE THREAD

**Issue by mcm1957** [AlCalzone/create-adapter#1083]
The standard workflow installed by create-adapter should either list master and main as triggering bransches or use * to run at all branches.

Why?
The workflow is often copied. And repositories switch from master to main later. In many cases the adaption of the workflow is missed and the fact that the workflow does no longer run is detected during repository review only.

As there is no negative sideeffect (at least as far as I know) listing both usual main branch names would be a low cost improvement. If using '*' to run the tests on co mmits to every branch would be an option but this should be discussed.

**Comment by AlCalzone** [comment]
Sounds reasonable

**Comment by mcm1957** [comment]
And the deploy code should ONLY be executed if master or main branch. Currently several users change branch to * which is ok if they really wnat to run tests on all branches. BUT it should be prohibited that a release is deployed which is not at main/master branch.

**Comment by AlCalzone** [comment]
**Why** should a tagged release not be made if the branch isn't master/main?

**Comment by mcm1957** [comment]
Iobroker environment / repository builder reads several information, i.e. io-package etc. directly from github main/master branch. Users are directed to README:md at master/main branch (head branch) whenever they click at the ? sign. So releases should (I would say must) be present at head branch. All checks are executed againts the head (master/main) branch (ok this could or even should be changed - but currently checks are executed against the head)

Deploying directly from a i.e. branch contains the risc that this branch is not merged into main/master afterwards and in a very shirt timeframe.

**Comment by AlCalzone** [comment]
One use case for releasing from a branch are backports, e.g. a fix for a 1.9 release when 2.0 is available and on master. That wouldn't work if we forbid the workflow from running.

**Comment by mcm1957** [comment]
Yes I see. don't know if npm allows lower version number at all - but I assume it does due to your use case.

But thats a ruther seldom case which can be simply workarounded by a manual npm deploy.
AND such a version will not be provided by iobroker latest repository (unless npm tags are set manually) anyway.
I do not remember that this use case occured within the last year.

In addition the standard workflow limits tests and deploys to main/master too (which is OK in my oppinion).

So I see more benefits (forgotten merge to master/main) than disadvantes (manual deploy for an action requiring many manual steps anyway.

But thats my oppinion and I can live with others too of course.

**Comment by AlCalzone** [comment]
I think the more flexible approach would be to warn in the release script when attempting to release from a branch that's not the default one.

**Comment by mcm1957** [comment]
OK, sound good too

**Comment by Apollon77** [comment]
newest release script has this, so nothing needed here?

**Comment by mcm1957** [comment]
This PR is related to TEST-AND-RELASE workflow TEMPLATE. This not realted to release scritp at all :-)

Gold is that all uptodate test-and-release workflows do not specify master OR main, bute react on both. This reduces problems when the head branch is later renamed and avoids porblems id test-and-release workflow is copied from template but no one checks if the head is main or master.

So the template still needs to be adjusted (if it was not changed already - I did not check)

The second part of this issue - filtering deploy code - is solved by check in releasescript.

**Comment by mcm1957** [comment]
@Apollon77 
Please REOPEN - closed on error.
